core/src/services/github/writer.rs

```diff
@@ -35,16 +35,47 @@ impl GithubWriter {
     pub fn new(core: Arc<GithubCore>, path: String) -> Self {
         GithubWriter { core, path }
     }
+
+    fn parse_metadata(headers: &http::HeaderMap, body: &[u8]) -> Result<Metadata> {
+        let mut metadata = Metadata::default();
+
+        if let Some(last_modified) = parse_last_modified(headers)? {
+            metadata.set_last_modified(last_modified);
+        }
+
+        if let Some(etag) = parse_etag(headers)? {
+            metadata.set_etag(etag);
+        }
+
+        if let Ok(body_str) = std::str::from_utf8(body) {
+            if let Ok(json) = serde_json::from_str::<serde_json::Value>(body_str) {
+                if let Some(content) = json.get("content") {
+                    if let Some(sha) = content.get("sha").and_then(|v| v.as_str()) {
+                        metadata.set_version(sha);
+                    }
+                } else if let Some(sha) = json.get("sha").and_then(|v| v.as_str()) {
+                    metadata.set_version(sha);
+                }
+            }
+        }
```

Comment:
Hi, `version` in opendal means that users can read like `op.read_with(path).version(version)`. Is it possibe for github service to do so? 

If it does, maybe we can have a strong typed response struct here so that we don't need to handle it at runtime.